src/index.js

```diff
@@ -18,21 +18,45 @@ const TCP = and(IP, base('tcp'))
 const UDP = and(IP, base('udp'))
 const UTP = and(UDP, base('utp'))
 
+const TCP_IPFS = or(
```

Comment:
Well, then you do a match for those and negate their values, this is all booleans anyway, you can compose things